Remove check for Jetpack and WCS from Stripe onboarding task

chickenn00dle opened the pull request:
This removes the check for Jetpack and WCS during the Stripe onboarding task.

This is part of the work to move Stripe OAuth code from WC Services over to Stipe while also removing the Jetpack dependency from the connection process.

### Detailed test instructions:

The changes in this PR are dependent on the following two PRs in [woocommerce-connect-server](https://github.com/Automattic/woocommerce-connect-server/pull/1443) and [woocommerce-gateway-stripe](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1267) checked out.

You can test this now, or wait for the Connect Server PR to be merged as it won't impact the current OAuth flow.

To test now, you will need to have both of the above installed/running locally. Be sure to set the `WOOCOMMERCE_CONNECT_SERVER_URL` constant set in your test site.

- With the above conditions, make sure you have a fresh installation of WordPress set up and WooCommerce is installed. (Make sure Jetpack has not been activated)
- Go through the onboarding wizard, being sure to skip the WCS/Jetpack installation step
- You should be able to complete the oauth flow for the Stripe Payment task with this branch checked out.

Note: When testing with a sandbox Stripe account, you will need to manually change instances of `publishable_key` and `secret_key` to `test_publishable_key` and `test_secret_key` in the following files:

-  client/task-list/tasks/payments/stripe.js
-  client/task-list/tasks/payments/methods.js

### Changelog Note:

- Enhancement: Setting up Stripe via the onboarding task list no longer requires WooCommerce Services or Jetpack to be installed.

chickenn00dle commented:
An important note, this should not be shipped until after Stripe Gateway changes. See pb0GrA-Pd-p2 for more details.

joshuatf commented:
Thanks for these changes, @chickenn00dle!  I'm going to set this PR to on-hold for now until those 2 PRs are merged.  I'll keep an eye on them, but if they land before I notice feel free to ping me for a review.

chickenn00dle commented:
I've pushed https://github.com/woocommerce/woocommerce-admin/pull/4933/commits/95265111eb1143fffbd276f2603ede897d093362 which adds a ToS prompt and link to revert to manual config in the event merchants want to opt out of sharing data: https://d.pr/i/HCXRsZ

chickenn00dle commented:
Hey @joshuatf, the other PRs associated with these changes have been merged and this is now ready for review 🙏

joshuatf commented on client/task-list/tasks/payments/stripe.js:
```diff
@@ -257,6 +247,42 @@ class Stripe extends Component {
+					/>
+				),
+				manualConfigLink: (
+					<Link
```
Minor, but I think semantically a `Button` with an `isLink` prop may be more appropriate here since there's no actual link.

chickenn00dle replied:
This makes sense. I've changed this in https://github.com/woocommerce/woocommerce-admin/pull/4933/commits/50a50b293f9a30c423c1db53ca09b391ff9f8ecb

I am seeing a change in color though: https://d.pr/i/jbpKDU

I think this is fine, but just double-checking whether there is a class we can add to get the color to match the `Link` component?

joshuatf replied:
Good eye.  The link buttons seem to be explicitly set to that color and regular links should probably be updated to match it (or vice versa), but we can tackle in a follow-up.

joshuatf reviewed (approved):
Thanks for this, @chickenn00dle!  This is a big improvement in the Stripe connection flow.

Left one small change, but this is testing well and pre-approving.

Pre-existing prior to this PR, but I found a small redundancy and missing link in the manual connection help text (See 2 instances of "Create One").  I'll open an issue to get this fixed.

<img width="712" alt="Screen Shot 2020-11-17 at 10 54 17 AM" src="https://user-images.githubusercontent.com/10561050/99415031-67cba600-28c5-11eb-9678-92bcc5789aa1.png">

chickenn00dle commented:
Thanks for the review @joshuatf!

> Pre-existing prior to this PR, but I found a small redundancy and missing link in the manual connection help text (See 2 instances of "Create One"). I'll open an issue to get this fixed.

Good catch! Thanks for handling this.

I've addressed the comment you left above. Let me know what you think, and once this is okay to merge.

Thank you!

joshuatf reviewed (approved):
Thanks for the edit!  This is testing well and LGTM :shipit:
